website/website/add_youtuber.py

```diff
@@ -0,0 +1,133 @@
+import os
+from urllib.parse import unquote, urlparse
+
+from googleapiclient.discovery import build
+
+
+class YoutubeApi:
+    """A class to interact with the YouTube API."""
+
+    #: (str): The API key used to authenticate requests to the YouTube API.
+    API_KEY = None
+
+    #: (str): The URL of the YouTube channel.
+    channel_url = None
+
+    #: (str): The custom name of the channel.
+    username = None
+
+    #: (str): The identifier of the channel.
+    channel_id = None
+
+    #: (str): The title of the channel.
+    channel_title = None
+
+    #: (str): The full description of the channel.
+    channel_description = None
+
+    def __init__(self, API_KEY, channel_url) -> None:
+        self.API_KEY = os.environ.get('API_KEY')
+        self.channel_url = channel_url
+        self.username = None
+        self.channel_id = None
+        self.channel_title = None
+        self.channel_description = None
+
+    def build_youtube_service(self):
+        """
+        Builds the YouTube service.
+        This method uses the Google API client library to build a service object for interacting
+        with the YouTube API.
+
+        Returns:
+        (googleapiclient.discovery.Resource): The service object for the YouTube API.
+
+        """
+        return build('youtube', 'v3', developerKey=self.API_KEY)
+
+    def get_snippet_using_username(self, youtube, username):
+        """
+        Gets the snippet data for a YouTube channel using the username.
+
+        Args:
+        youtube (googleapiclient.discovery.Resource): The service object YouTube object.
+        username (str): The username of the YouTube channel.
+
+        Returns:
+        (dict): The snippet data of the YouTube channel.
+
+        """
+        return youtube.search().list(
+            q=username,
+            part="snippet",
+            type="channel",
+            maxResults=1
+        )
+
+    def get_snippet_using_id(self, youtube, channel_id):
+        """
+        Gets the snippet data for a YouTube channel using the channel ID. Can return more full data.
+
+        Args:
+        youtube (googleapiclient.discovery.Resource): The service object YouTube object.
+        channel_id (str): The ID of the YouTube channel.
+
+        Returns:
+        (dict): The snippet data for the YouTube channel.
+
+        """
+        return youtube.channels().list(
+            part="snippet",
+            id=channel_id
+        )
+
+    def get_username_from_url(self, channel_url: str) -> str:
```

Comment:
I'd consider making the methods that are not expected to be called outside of the class to be private.

get_username_from_url -> __get_username_from_url, if you are not going to address to the method outside of this class.

Same to all the class attributes.
If you do not need to access them outside of the class -> make them private.